### PR TITLE
Mtdsim-103 Strategy DropDown

### PR DESCRIPTION
**Pull request by FearMyCode**
- Refactored the 'DropDown' component to support multi-selection of strategies
- Strategy options are populated based on the provided scheme.
- Added the 'Chip' component
- Pass strategies array to the backend
- Error handling mechanisms for strategy selection
- Updated strategies for scenarios

Backend:
print(scheme)
print(strategies)

alternative
['IPShuffle', 'ServiceDiversity']

simultaneous
['CompleteTopologyShuffle', 'IPShuffle', 'OSDiversity', 'ServiceDiversity']

### CODE REVIEW

File: client/src/components/Chip.vue [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    class="inline-flex bg-gray-200 rounded-full px-3 py-px justify-center content-center cursor-default"
+  >
+    <span class="truncate block max-w-xs align-middle text-xs">{{
+      label
+    }}</span>
+    <div class="cursor-pointer">
+      <svg-icon
+        type="mdi"
+        size="20"
+        :path="mdiCloseCircleOutline"
```
Comment (benvarian) [inline client/src/components/Chip.vue:12]: use a different icon

Comment (benvarian) [reply client/src/components/Chip.vue:12]: add some form of padding to the gap between the word and the X

File: client/src/components/Chip.vue [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<template>
+  <div
+    class="inline-flex bg-gray-200 rounded-full px-3 py-px justify-center content-center cursor-default"
```
Comment (benvarian) [inline client/src/components/Chip.vue:3]: change the background to be not gray-200

File: client/src/components/DropDown.vue [hunk 3]
```diff
@@ -2,19 +2,27 @@
   <label>{{ label }}</label>
   <div class="relative">
     <div
-      class="p-1 mt-2 pl-2.5 border border-solid rounded-md text-black w-full bg-white hover:cursor-pointer"
+      class="p-1 mt-2 border border-solid rounded-md text-black w-full bg-white hover:cursor-pointer hover:border-gray-400"
```
Comment (benvarian) [inline client/src/components/DropDown.vue:5]: change the hover bg to not be the same as the active bg

File: client/src/components/DropDown.vue [hunk 4]
```diff
@@ -25,63 +33,85 @@
...
+          <li
+            v-for="(item, index) in menuOptions"
+            :key="index"
+            class="px-4 py-2 text-sm text-black leading-5 hover:bg-slate-300 hover:text-black focus:outline-none focus:bg-gray-50 hover:cursor-pointer"
+            @click="handleClick(item)"
+            :class="{
+              'hover:cursor-not-allowed text-black hover:text-black':
+                item != 'Random',
+              'font-bold': isSelected(item),
+              'bg-slate-300': isSelected(item),
```
Comment (benvarian) [inline client/src/components/DropDown.vue:50]: use another color than slate

File: client/src/components/DropDown.vue [hunk 5]
```diff
@@ -25,63 +33,85 @@
...
+        <ul>
+          <li
+            v-for="(item, index) in menuOptions"
+            :key="index"
+            class="px-4 py-2 text-sm text-black leading-5 hover:bg-slate-300 hover:text-black focus:outline-none focus:bg-gray-50 hover:cursor-pointer"
+            @click="handleClick(item)"
+            :class="{
+              'hover:cursor-not-allowed text-black hover:text-black':
+                item != 'Random',
+              'font-bold': isSelected(item),
```
Comment (benvarian) [inline client/src/components/DropDown.vue:49]: doubling up on class calls

File: client/src/components/DropDown.vue [hunk 6]
```diff
@@ -25,63 +33,85 @@
...
-  menuOptions: {
-    type: Array,
-    default: () => [],
-  },
-  modelValue: { type: String, default: "" },
+  menuOptions: { type: Array, default: () => [] },
+  modelValue: { type: [Array, String], default: () => [] },
   error: { type: String, default: "" },
+  multiSelect: { type: Boolean, default: false },
+  maxSelection: { type: Number, default: Infinity },
```
Comment (benvarian) [inline client/src/components/DropDown.vue:75]: doesn't having the default be set to infinity seem a bit silly

File: client/src/components/DropDown.vue [hunk 7]
```diff
@@ -25,63 +33,85 @@
...
+            class="px-4 py-2 text-sm text-black leading-5 hover:bg-slate-300 hover:text-black focus:outline-none focus:bg-gray-50 hover:cursor-pointer"
+            @click="handleClick(item)"
+            :class="{
+              'hover:cursor-not-allowed text-black hover:text-black':
+                item != 'Random',
+              'font-bold': isSelected(item),
+              'bg-slate-300': isSelected(item),
+            }"
+          >
+            {{ item }}
```
Comment (benvarian) [inline client/src/components/DropDown.vue:53]: i assume that this is the item that the user can click on, there is text sizing problems so pls fix that

File: server/api/app.py
Comment (nicodeas) [inline server/api/app.py:1]: This has been changed on main. 
Please check

File: client/src/components/DropDown.vue [hunk 8]
```diff
@@ -2,19 +2,27 @@
   <label>{{ label }}</label>
```
Comment (benvarian) [inline client/src/components/DropDown.vue:2]: the padding on the left doesnt match up with the other form fields

Comment (benvarian) [reply client/src/components/DropDown.vue:2]: also if i select random in the scheme drop down im still able to pick a strat

**Comment by benvarian** [comment]
im writing in comments cause i cant be bothered to find the stuff
- fix the selected background to not be gray if your gonna use aqua??
- this aqua color that your using isnt the same as the buttons up top
- the padding in the selected strategy is wrong, the text and the x button are on a dif x axis i guess
- if i select random in the scheme i shouldnt be able to click any just like none

**Review by nicodeas — APPROVED** [review]

**Pull request merged**